lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
PS - you said you use PG, but don't use pgbouncer? You need to get on that!
haha -- trust me, you'll love it. The performance increase is insane. :1:

On Thu, Apr 2, 2015 at 12:56 PM, Kyle Renfrow <frow05@gmail.com> wrote:

> I feel ya. So we are in agreement.
>
> Ie, set names is the best way to set the encoding BUT, is it worth the
> overhead?
>
> Note: we could leave it as all the user has to do is NOT SET the charset
> in their connection params and it's skipped.
>
> Note, this email was sent from my cell phone.
> On Apr 2, 2015 12:53 PM, "Steve Müller" <notifications@github.com> wrote:
>
>> In lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php
>> <https://github.com/doctrine/dbal/pull/828#discussion_r27675360>:
>>
>> > @@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
>> >                  $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
>> >              }
>> >
>> > +            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
>> > +             * - the 'client_encoding' connection param only works with postgres >= 9.1
>> > +             * - passing client_encoding via the 'options' param breaks pgbouncer support
>> > +             */
>> > +            if (isset($params['charset'])) {
>> > +              $pdo->query('SET NAMES \''.$params['charset'].'\'');
>>
>> Yep I added it back then but I wasn't aware of the pgbouncer thingy...
>> Well other drivers support setting the charset explicitly, too so
>> supporting this for pgsql is reasonable. TBH I COULD live with SET NAMES
>> I am just curious on other opinions about this...
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/doctrine/dbal/pull/828/files#r27675360>.
>>
>